docs/concepts/macros/sqlmesh_macros.md

```diff
@@ -911,6 +911,37 @@ The methods are available because the `column` argument is parsed as a SQLGlot [
 
 Column expressions are sub-classes of the [Condition class](https://sqlglot.com/sqlglot/expressions.html#Condition), so they have builder methods like [`between`](https://sqlglot.com/sqlglot/expressions.html#Condition.between) and [`like`](https://sqlglot.com/sqlglot/expressions.html#Condition.like).
 
+#### Accessing model schemas
+
+Model schemas can be accessed within a Python macro function through its evaluation context's `column_to_types` method, when they can be statically determined.
+
+For example, consider the following macro function which aims to rename the columns of a target model by adding a prefix to them:
+
+```python linenums="1"
+from sqlglot import exp
+from sqlmesh.core.macros import macro
+
+@macro()
+def foo(evaluator, model_name, prefix):
+    prefix = prefix.name
+    renamed_projections = []
+
+    for name, dtype in evaluator.columns_to_types(model_name.sql()).items():
+        new_name = prefix + name
+        renamed_projections.append(exp.cast(exp.column(name), dtype).as_(new_name))
+
+    return renamed_projections
+```
+
+Having access to the schema of an upstream model can be useful for various reasons:
+
+- Normalizing columns so that downstream consumers are not tightly coupled e.g. to external or source tables
+- Projecting only a subset of columns that satisfy some criteria
+- Adding necessary casts
+- Computing common columns
```

Comment:
What does this mean?

Reply:
You can combine columns from upstream models, e.g. by adding them, thus computing new columns. We should probably improve this section a bit and add more examples, I understand that it's a bit hand-wavy at the moment.

@z3z1ma any practical use-case comes to mind from your experience?

Reply:
if you do rule based naming on an upstream column, for example `n_<column>` always corresponds to a numeric -- you can autocompute statistics like a pandas describe. And subsequent `plans` will simply include them by virtue of updating the external model for example. very dry.

Some columns can be determined to be PII based on name and a blanket hash may be sufficient for any column containing `email` or `address`, etc.

Reply:
Actually even better, you can use the `type` 😆 
So generating summary tables that are always in sync with upstream source may be cake.

Reply:
Perfect, incorporated the statistics example in the latest commit